File: 1979-find-greatest-common-divisor-of-array/1979-find-greatest-common-divisor-of-array.java
class Solution {
    public int findGCD(int[] nums) {

        Arrays.sort(nums);

        int min = nums[0];
        int max = nums[nums.length-1];

        return gdc(min, max);
    }

    int gdc(int a, int b) { //최대 공약수

        if(a<b) // 유클리드 호제법 조건
        {
            int temp = a;
            a = b;
            b = temp;
        }
        while(b!=0) { // 유클리드 호제법
            int r=a%b;
            a=b;
            b=r;
        }

        return a;
    }
    
}

File: 1979-find-greatest-common-divisor-of-array/1979-find-greatest-common-divisor-of-array.js
/**
 * @param {number[]} nums
 * @return {number}
 */
var findGCD = function(nums) {
    console.log(nums.sort((a,b)=>a-b))
    console.log(nums[0],nums[nums.length-1])
    let max=nums[nums.length-1]
    let min=nums[0]
    
    while(1){
        let r=max%min;
        if(r==0) return min
        
        max=min;
        min=r;
    }
};